### DISCUSSION TITLE
[Termux] How to tell box64 use amd64 wine ?

### DISCUSSION THREAD

**Discussion by xDoge26** [ptitSeb/box64#435] [Q&A]
i downloaded wine and extracted it
But i dont know how to make box64 use amd64 wine
```cd
mkdir wine64
cd wine64
wget https://www.playonlinux.com/wine/binaries/phoenicis/upstream-linux-amd64/PlayOnLinux-wine-6.17-upstream-linux-amd64.tar.gz
tar -xvf *.tar.gz
cd

**Comment by ptitSeb** [comment]
Take a look at this https://box86.org/2022/02/box86-box64-are-easy-to-use/ , does that help?

Also, don't forget wine64 still have 32bits apps, and so needs box86 too.

**Reply by xDoge26** [reply]
> Take a look at this https://box86.org/2022/02/box86-box64-are-easy-to-use/ , does that help?
> 
> Also, don't forget wine64 still have 32bits apps, and so needs box86 too.

will this method work too ?
i searched google and found this

```
Add these lines to your /etc/profile:
export BOX86_PATH=~/wine/bin/
export BOX86_LD_LIBRARY_PATH=~/wine/lib/wine/i386-unix/:/lib/i386-linux-gnu:/lib/aarch64-linux-gnu/
export BOX64_PATH=~/wine64/bin/
export BOX64_LD_LIBRARY_PATH=~/wine64/lib/wine/i386-unix/:~/wine64/lib/wine/x86_64-unix/:/lib/i386-linux-gnu/:/lib/x86_64-linux-gnu:/lib/aarch64-linux-gnu

**Comment by ptitSeb** [comment]
you don't need the BOX LD_LIBRARY_PATH
the BOX86_PATH / BOX64_PATH are only usefull for stuff launched from box86 / box64. If you want to use `wine` directly, you need to use regular `PATH`.
You also need to have binfmt system integration

**Reply by xDoge26** [reply]
> you don't need the BOX LD_LIBRARY_PATH the BOX86_PATH / BOX64_PATH are only usefull for stuff launched from box86 / box64. If you want to use `wine` directly, you need to use regular `PATH`. You also need to have binfmt system integration

how do i install binfmt

**Comment by ptitSeb** [comment]
Have tried `sudo make install` ? Did you read the link I provided?

**Reply by xDoge26** [reply]
> Have tried `sudo make install` ? Did you read the link I provided?

you mean this ?
`sudo apt install libgtk2.0-0:armhf libsdl2-image-2.0-0:armhf`
sorry im noob

**Reply by ptitSeb** [reply]
No, I'm talking about compiling and installing box64 and box86.

**Reply by xDoge26** [reply]
> No, I'm talking about compiling and installing box64 and box86.

i already compiled and installed both box86 and box64

**Reply by ptitSeb** [reply]
then just run wine. `wine64/bin/winefg` for example.

**Reply by xDoge26** [reply]
> then just run wine. `wine64/bin/winefg` for example.

i dont really understand ,.If i want to run 64bit 7z.exe  , it will be `box64 wine64/bin` then what

**Reply by ptitSeb** [reply]
you should have box64 binfmt integration working if you are on a regular debian/ubuntu system, so you don't need to prefix the command "box64"

to run 7z.exe, just use `~/wine64/bin/wine64 7z.exe` if you expanded your wine in wine64 folder in your home.

**Reply by xDoge26** [reply]
> you should have box64 binfmt integration working if you are on a regular debian/ubuntu system, so you don't need to prefix the command "box64"
> 
> to run 7z.exe, just use `~/wine64/bin/wine64 7z.exe` if you expanded your wine in wine64 folder in your home.

then how do i run 32-64bit 7z.exe without `~/wine64/bin/wine64` where do i move x86-64 wine to

**Reply by ptitSeb** [reply]
Then why didn't you just follow the instructions I gave on the box86 blog post?

**Reply by ptitSeb** [reply]
Yes, that will create symlink so you can just use `wine 7z.exe` or `wine64 mygame.exe` (but you need to have extracted wine in the wine folder of you home, not wine64).

**Comment by xDoge26** [comment]


**Reply by ptitSeb** [reply]
That's because you don't have binfmt integration. It's not available on termux (it can be enable it seems, but I don't know how, and it's not a box issue).
You need to prefix everything with `box86` or `box64`. Or, you can run setup x86_64 bash, that should help. Look here: https://box86.org/2022/09/running-bash-with-box86-box64/ for more details.

**Reply by raspiduino** [reply]
Some Android does not have binfmt enabled by default. You can enable it by manually compile the kernel.
